Library/Homebrew/cask/cask.rb

```diff
@@ -231,6 +231,11 @@ def eql?(other)
     alias == eql?
 
     def to_h
+      if loaded_from_api && Homebrew::EnvConfig.install_from_api?
+        json_cask = Homebrew::API::Cask.all_casks[token]
+        return Homebrew::API.merge_variations(json_cask)
+      end
```

Comment:
I'd rather not have to add this code to each of the `#to_h*` methods. Is this functionally necessary or a performance improvement? Put another way we'll have already loaded the cask/formula from the API at this point; will the normal `#to_h*` method return the expected output.

Reply:
If it's a performance optimization, it'd be nice to have benchmarks. Also, if it's only relevant for `brew info` I'd rather add the code there instead of to the `#to_h*` methods.

Reply:
> Put another way we'll have already loaded the cask/formula from the API at this point; will the normal `#to_h*` method return the expected output?

The original implementation was non-functional: https://github.com/Homebrew/brew/issues/14541

Reply:
Ah okay, that was the context I was missing. Looks like the `*flight` blocks are still a thorn in our side when it comes to the cask API and that's really what needs to be fixed eventually but I guess this is fine as a temporary fix.

Since we aren't using the `*flight` block info right now when installing from the API (we download the source instead) we could probably just revert back to the previous behavior instead of bundling each block as a string. It might actually solve this problem. I could probably get a PR related to that up by tomorrow.

Reply:
> Since we aren't using the *flight block info right now when installing from the API (we download the source instead) we could probably just revert back to the previous behavior instead of bundling each block as a string. It might actually solve this problem. I could probably get a PR related to that up by tomorrow.

@apainintheneck This sounds good 👍🏻